### PR TITLE
ConfigProperties and Storage for demo

### PR DESCRIPTION
**Pull request by sanyavertolet**
### What's done:
 * Added `ToolKey`
 * Implemented `ToolStorage`
 * Updated `application.properties`
 * Added `application-mac.properties`
 * Added `save-demo` configuration

### CODE REVIEW

File: save-demo/src/main/resources/application-mac.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+demo.fileStorage.location=/Users/Shared/.save-cloud/cnb/files
```
Comment (petertrr) [inline save-demo/src/main/resources/application-mac.properties:1]: BTW, what's wrong with `user.home` on Mac?

**Review by petertrr — APPROVED** [review]

File: save-demo/src/main/resources/application-mac.properties (hunk 1 above)
Comment (sanyavertolet) [reply save-demo/src/main/resources/application-mac.properties:1]: I do not want to move existing .save_cloud from /Users/Shared to user.home

File: save-demo/src/main/resources/application.properties [hunk 2]
```diff
@@ -1,3 +1,4 @@
 server.port=5401
 management.endpoints.web.exposure.include=health,info,prometheus
 spring.codec.max-in-memory-size=100MB
+demo.fileStorage.location=${user.home}/.save-cloud/cnb/files
```
Comment (nulls) [inline save-demo/src/main/resources/application.properties:4]: what's about prod configuration?

Comment (nulls) [reply save-demo/src/main/resources/application.properties:4]: yeap, but I'm asking more about using `${user.home}` in prod

Comment (nulls) [reply save-demo/src/main/resources/application.properties:4]: sandbox and backend use `/home/cnb/files` as file storage location. better to use the similar location

Comment (sanyavertolet) [reply save-demo/src/main/resources/application.properties:4]: Will do it as a part of #1450

Comment (sanyavertolet) [reply save-demo/src/main/resources/application.properties:4]: Done

**Pull request merged**